### DISCUSSION TITLE
Pygimli 1.4.0 and tetgen

### DISCUSSION THREAD

**Discussion by Morriiiss** [gimli-org/pyGIMLi#526] [General]
Hi I have been using Pygimli 1.3.1 for a while and I recently updated to pygimli 1.4.0 and noticed that I got error when trying to run ert 3d inversion. I tried on 3 different machines and all got the same error, the solution was to separate install tetgen (conda install tetgen) because that was missing. I cant remember I had to do this in the previous version of pygimli.

I just wanted to bring this up, maybe it should be this way, I don't know.

Really great and cool package you guys have made, much appreciated.

**Comment by halbmy** [comment]
I guess this happened on Windows? Indeed it seems like tetgen is not a dependency for the conda package which should be corrected.

Thank you for reporting.

**Comment by halbmy** [comment]
Well, actually tetgen is a requirement for the pyGIMLi conda package, no matter which platform you are working on. Can you please tell, how you installed pyGIMLi?

**Comment by Morriiiss** [comment]
Hey,

We were three people installing PyGimli independently of each other with
following the instructions on the PyGimli website,
"conda create -n pg -c gimli -c conda-forge pygimli=1.4.0"
We all got that tetgen were missing, never happened before on the prev
version.
We installed it on windows and on WSL 2 running Debian and all the systems
got the same error.

Den tis 2 maj 2023 kl 10:26 skrev Thomas Günther ***@***.***>:

**Comment by halbmy** [comment]
Yes, you are absolutely right. Both on Linux and Windows, `pygimli=1.4` does not bring tetgen as dependency, whereas `pygimli=1.3` did. We need to correct this. Thank you for reporting.

**Comment by halbmy** [comment]
I am not sure on which basis I wrote the latter comment, but current pyGIMLi versions (1.4.1 and 1.4.2) do bring tetgen as a dependency. I tested this on several platforms:
```
  pgcore             gimli/win-64::pgcore-1.4.0-py39_0
  pillow             conda-forge/win-64::pillow-9.5.0-py39ha9166d5_1
  pip                conda-forge/noarch::pip-23.1.2-pyhd8ed1ab_0
  platformdirs       conda-forge/noarch::platformdirs-3.6.0-pyhd8ed1ab_0
  pluggy             conda-forge/noarch::pluggy-1.0.0-pyhd8ed1ab_5
  ply                conda-forge/noarch::ply-3.11-py_1
  pooch              conda-forge/noarch::pooch-1.7.0-pyha770c72_3
  proj               conda-forge/win-64::proj-9.2.0-heca977f_0
  pthread-stubs      conda-forge/win-64::pthread-stubs-0.4-hcd874cb_1001
  pthreads-win32     conda-forge/win-64::pthreads-win32-2.9.1-hfa6e2cd_3
  pugixml            conda-forge/win-64::pugixml-1.11.4-h63175ca_1
  pygimli            gimli/noarch::pygimli-1.4.2-py_0
  pyparsing          conda-forge/noarch::pyparsing-3.1.0-pyhd8ed1ab_0
  pyqt               conda-forge/win-64::pyqt-5.15.7-py39hb77abff_3
  pyqt5-sip          conda-forge/win-64::pyqt5-sip-12.11.0-py39h99910a6_3
  pysocks            conda-forge/noarch::pysocks-1.7.1-pyh0701188_6
  pytest             conda-forge/noarch::pytest-7.3.2-pyhd8ed1ab_1
  python             conda-forge/win-64::python-3.9.16-h4de0772_0_cpython
  python-dateutil    conda-forge/noarch::python-dateutil-2.8.2-pyhd8ed1ab_0
  python_abi         conda-forge/win-64::python_abi-3.9-3_cp39
  pyvista            conda-forge/noarch::pyvista-0.39.1-pyhd8ed1ab_0
  qt-main            conda-forge/win-64::qt-main-5.15.8-h2c8576c_12
  requests           conda-forge/noarch::requests-2.31.0-pyhd8ed1ab_0
  scipy              conda-forge/win-64::scipy-1.10.1-py39hde5eda1_3
  scooby             conda-forge/noarch::scooby-0.7.2-pyhd8ed1ab_0
  setuptools         conda-forge/noarch::setuptools-67.7.2-pyhd8ed1ab_0
  sip                conda-forge/win-64::sip-6.7.9-py39h99910a6_0
  six                conda-forge/noarch::six-1.16.0-pyh6c4a22f_0
  snappy             conda-forge/win-64::snappy-1.1.10-hfb803bf_0
  sqlite             conda-forge/win-64::sqlite-3.42.0-hcfcfb64_0
  suitesparse        conda-forge/win-64::suitesparse-5.4.0-h5d0cbe0_1
  svt-av1            conda-forge/win-64::svt-av1-1.4.1-h63175ca_0
  tbb                conda-forge/win-64::tbb-2021.9.0-h91493d7_0
  tbb-devel          conda-forge/win-64::tbb-devel-2021.9.0-h91493d7_0
  tetgen             conda-forge/win-64::tetgen-1.6.0-ha925a31_0
```

Can you please check again?
